client/components/marketing-survey/cancel-jetpack-form/index.jsx

```diff
@@ -0,0 +1,258 @@
+			this.recordEvent( 'calypso_purchases_cancel_form_start' );
+		}
+	}
+
+	/**
+	 * Get possible steps for the survey
+	 */
+	getAvailableSurveySteps() {
+		// this could vary based on the user data/ product
+		// for now, only showing two steps for all Jetpack plans and products
+		return [ steps.FEATURES_LOST_STEP, steps.CANCELLATION_REASON_STEP ];
+	}
+
+	// Record an event for Tracks
+	recordEvent = ( name, properties = {} ) => {
+		const { purchase, flowType, isAtomicSite } = this.props;
+
+		this.props.recordTracksEvent( name, {
+			cancellation_flow: flowType,
+			product_slug: purchase.productSlug,
+			is_atomic: isAtomicSite,
+
+			...properties,
+		} );
+	};
+
+	/**
+	 * Set the cancellation flow back to the beginning
+	 * Clear out stored state for the flow
+	 */
+	resetSurveyState() {
+		this.setState( {
+			cancellationStep: steps.FEATURES_LOST_STEP,
+			surveyResponse: false,
+		} );
+	}
+
+	handleCloseDialog = () => {
+		this.props.onClose();
+		this.resetSurveyState();
+
+		// record tracks event
+		// sends the same event name as the main product cancellation form
+		this.recordEvent( 'calypso_purchases_cancel_form_close' );
+	};
+
+	setSurveyStep = ( stepFunction ) => {
+		const availableSteps = this.getAvailableSurveySteps();
+		const newStep = stepFunction( this.state.cancellationStep, availableSteps );
+
+		this.setState( {
+			cancellationStep: newStep,
+		} );
+
+		// record tracks event
+		// since the steps used for the Jetpack cancellation flow are different, use a different event name for Tracks
+		this.recordEvent( 'calypso_purchases_cancel_jetpack_survey_step', { new_step: newStep } );
+	};
+
+	clickNext = () => {
+		this.setSurveyStep( nextStep );
+	};
+
+	clickPrevious = () => {
+		this.setSurveyStep( previousStep );
+	};
+
+	onSubmit = () => {
+		// TODO: handle survey submission - probably using the existing submitSurvey method
+
+		// call back to the parent component to actually cancel the subscription
+		this.props.onClickFinalConfirm();
+
+		// record tracks event
+		// this uses the same event name as the main product cancellation form
+		// this way, all cancellation stats can be viewed together
+		this.recordEvent( 'calypso_purchases_cancel_form_submit' );
+	};
+
+	/**
+	 * Render the dialog buttons for the current step
+	 */
+	renderStepButtons() {
+		const { disableButtons, translate } = this.props;
+		const disabled = disableButtons;
+
+		// in most cases, we will have a previous and next
+		const close = {
+			action: 'close',
+			disabled,
+			label: translate( "I'll Keep It" ),
+		};
+		const next = {
+			action: 'next',
+			disabled: disabled,
+			label: translate( 'Continue Cancelling' ),
+			onClick: this.clickNext,
+		};
+		const prev = {
+			action: 'prev',
+			disabled,
+			label: translate( 'Previous Step' ),
+			onClick: this.clickPrevious,
+		};
+
+		const cancelText = translate( 'Cancel Now' );
+		const cancellingText = translate( 'Cancelling' );
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'Why cancelling?' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/improve-jetpack-cancellation-flow)